src/Lykke.Job.BlockchainCashinDetector.AzureRepositories/DepositBalanceDetectionsDeduplicationRepository.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AzureStorage;
+using AzureStorage.Tables;
+using Common.Log;
+using Lykke.Job.BlockchainCashinDetector.Core.Domain;
+using Lykke.SettingsReader;
+
+namespace Lykke.Job.BlockchainCashinDetector.AzureRepositories
+{
+    public class DepositBalanceDetectionsDeduplicationRepository : IDepositBalanceDetectionsDeduplicationRepository
+    {
+        private readonly INoSQLTableStorage<DepositBalanceDetectionsDeduplicationEntity> _storage;
+
+        public static IDepositBalanceDetectionsDeduplicationRepository Create(IReloadingManager<string> connectionString, ILog log)
+        {
+            var storage = AzureTableStorage<DepositBalanceDetectionsDeduplicationEntity>.Create(
+                connectionString,
+                "DepositBalanceDetectionsDeduplication",
+                log);
+
+            return new DepositBalanceDetectionsDeduplicationRepository(storage);
+        }
+
+        private DepositBalanceDetectionsDeduplicationRepository(INoSQLTableStorage<DepositBalanceDetectionsDeduplicationEntity> storage)
+        {
+            _storage = storage;
+        }
+
+        public async Task<IEnumerable<IDepositBalanceDetectionsDeduplicationLock>> GetAsync(IEnumerable<IDepositWalletKey> keys)
+        {
+            return await _storage.GetDataAsync(keys.Select(x => new Tuple<string, string>
+                (
+                    DepositBalanceDetectionsDeduplicationEntity.GetPartitionKey(x.BlockchainType, x.BlockchainAssetId, x.DepositWalletAddress),
+                    DepositBalanceDetectionsDeduplicationEntity.GetRowKey(x.DepositWalletAddress)
+                )
+            ));
+        }
+
+        public async Task InsertOrReplaceAsync(string blockchainType, string blockchainAssetId, string depositWalletAddress, long block)
+        {
+            var entity = new DepositBalanceDetectionsDeduplicationEntity
```

Comment:
This is very unlikely, but still possible situation, when transaction completion event with lower block number could be processed after higher block number, which would lead to the balance duplication. This can be happen due to errors in another code and this is shouldn't be excluded. To handle this case correctly, you should check saved block number first and replace it, only if it older then processed one.

To do this, you can use ```InsertOrModifyAsync()``` method of the storage.